### PR TITLE
xDefi BTC-like support

### PR DESCRIPTION
**Pull request by nrjshka**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@viaprotocol/web3-wallets",
-  "version": "1.0.23",
+  "version": "1.0.24-rc.1",
```
Comment (insulineru) [inline package.json]: If PR ready – remove RC pls :)

File: src/context/WalletProvider.tsx [hunk 2]
```diff
@@ -379,19 +390,23 @@ const WalletProvider = function WalletProvider({ children }: { children: React.R
...
+        await provider.enable(chainList)
 
-        const offlineSigner = provider.getOfflineSigner(currentChain)
+        const offlineSigner = provider.getOfflineSigner(currentChain.network)
         const addressesList = await offlineSigner.getAccounts()
         const { address } = addressesList[0]
         const addressShort = shortenAddress(address)
         const connectedWallets = await getCosmosConnectedWallets(provider)
-        const addresesInfo = connectedWallets.reduce((acc, { addresses, chainId }) => ({ ...acc, [addresses[0]]: [chainId] }), {})
+        const addresesInfo = getAddresesInfo(connectedWallets)
```
Comment (insulineru) [inline src/context/WalletProvider.tsx:413]: 👍🏻

File: src/context/WalletProvider.tsx [hunk 3]
```diff
@@ -719,8 +734,21 @@ const WalletProvider = function WalletProvider({ children }: { children: React.R
...
+        const walletInfo = getWalletInfoByChainId(fromChainId)
+        const provider = currentState.provider.getProviderByChainId(fromChainId)
+
+        if (!walletInfo || !provider) {
+          throw new Error(`[Wallet] sendTx error: no wallet info for chainId ${fromChainId}`)
+        }
+
+        const result = await provider.transfer(transaction as BTClikeTransaction)
+
+        console.log('result!', result)
```
Comment (insulineru) [inline src/context/WalletProvider.tsx]: remove console pls

File: src/provider/xDeFi/xDeFi.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
+import type { Web3Provider } from '@coinbase/wallet-sdk/dist/provider/Web3Provider'
+import type { TProviderType, TXDeFiAvailableProviders } from './types'
+import { XDeFiProvider } from './xDeFiProvider'
+import { EVM_CHAINS, NETWORK_IDS } from '@/constants'
+import type { TXDeFiWeb3Provider } from '@/context'
+
+class XDeFi {
```
Comment (insulineru) [inline src/provider/xDeFi/xDeFi.ts:7]: image

File: src/utils/btc.ts [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import type { TChainWallet } from '..'
+import { btcChainWalletMap } from '..'
+import { getConnectedWallets } from './common'
+import type { XDeFi } from '@/provider'
+
+const getBTCConnectedWallets = async (xDeFiProvider: XDeFi) => {
+  return await getConnectedWallets(btcChainWalletMap, async ({ name }: TChainWallet) => {
+    return await xDeFiProvider.getProviderByKey(name as any).getAccounts()
```
Comment (insulineru) [inline src/utils/btc.ts]: as far as I know, you don't need to use `await` in `return`. Webstorm usually warns about this

**Review by insulineru — APPROVED** [review]

File: src/constants.ts [hunk 6]
```diff
@@ -74,42 +74,60 @@ export const ERRCODE = {
...
 
-export const cosmosChainsMap: { [key in typeof COSMOS_CHAINS[number]]: string } = {
-  [NETWORK_IDS.Cosmos]: 'cosmoshub-4',
-  [NETWORK_IDS.Osmosis]: 'osmosis-1',
-  [NETWORK_IDS.Sifchain]: 'sifchain-1'
-}
+export const isEvmChain = (chainId: number) => chainId > 0
+export const isCosmosChain = (chainId: number) => COSMOS_CHAINS.includes(chainId as any)
+export const isSolChain = (chainId: number) => SOL_CHAINS.includes(chainId as any)
+export const isBTCChain = (chainId: number) => BTC_CHAINS.includes(chainId as any)
```
Comment (insulineru) [inline src/constants.ts]: May we can rename to `isBTCLikeChain`?

File: src/utils/btc.ts (hunk 5 above)
Comment (nrjshka) [reply src/utils/btc.ts]: ty!

File: src/constants.ts (hunk 6 above)
Comment (nrjshka) [reply src/constants.ts]: sure!

**Pull request merged**